tests/Unit/DataStructures/Test_TensorMagnitude.cpp

```diff
@@ -0,0 +1,97 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+
+TEST_CASE("Unit.DataStructures.Tensor.EuclideanMagnitude",
+          "[DataStructures][Unit]") {
+  const size_t npts = 8;
+  DataVector one(npts, 1.0);
+  DataVector two(npts, 2.0);
+  DataVector minus_three(npts, -3.0);
+  DataVector four(npts, 4.0);
+  DataVector minus_five(npts, -5.0);
+  DataVector twelve(npts, 12.0);
+  DataVector thirteen(npts, 13.0);
+
+  tnsr::i<DataVector, 1, Frame::Grid> one_d_covector;
+  one_d_covector.get<0>() = two;
+  DataVector magnitude_one_d_covector = magnitude(one_d_covector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(magnitude_one_d_covector[s] == 2.0);
+  }
+
+  tnsr::A<DataVector, 1, Frame::Grid> one_d_vector;
+  one_d_vector.get<0>() = minus_three;
+  one_d_vector.get<1>() = four;
+  DataVector magnitude_one_d_vector = magnitude(one_d_vector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(magnitude_one_d_vector[s] == 5.0);
+  }
+
+  tnsr::I<DataVector, 2, Frame::Grid> two_d_vector;
+  two_d_vector.get<0>() = minus_five;
+  two_d_vector.get<1>() = twelve;
+  DataVector magnitude_two_d_vector = magnitude(two_d_vector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(magnitude_two_d_vector[s] == 13.0);
+  }
+
+  tnsr::i<DataVector, 3, Frame::Grid> three_d_covector;
```

Comment:
There is a constructor for `tnsr::vector` that takes a list of `DataVector`s that could be used here. You may need three braces, unfortunately, to get it to work.

If so, you can `const`